smallrye-reactive-messaging-amqp/src/main/java/io/smallrye/reactive/messaging/amqp/AmqpMessage.java

```diff
@@ -87,8 +86,15 @@ public AmqpMessage(io.vertx.amqp.AmqpMessage msg) {
 
     @Override
     public CompletionStage<Void> ack() {
-        this.message.accepted();
-        return CompletableFuture.completedFuture(null);
+        // We must switch to the context having created the message.
+        // This context is passed when this instance of message is created.
+        // It's more a Vert.x AMQP client issue which should ensure calling `accepted` on the right context.
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        this.context.runOnContext(x -> {
+            this.message.accepted();
+            future.complete(null);
+        });
```

Comment:
Should it check it needs to do this? Doesn't this also rather change the behaviour anywhere ack() was already being called on the correct thread?

Reply:
Do you mean implementing a trampoline? 

Hopefully, this is a temporary fix and the Vert.x AMQP client would do it more consistently.

Reply:
I just mean that if the ack method was already being used in other places on the correct event loop thread then it would have actually done the work when called previously, whereas now it will only do it later, possibly after whatever the thread attempts next once it has called ack. The difference may or may not be important. I'm mainly just surprised to see that change rather than only the off-event-loop handling.

Reply:
So you mean implementing a trampoline to check if we are on the correct context, and if not doing this logic. It can be done. I would have preferred having it done in the vert.x amqp client than here, as this is just a temporary fix for the real bug (which is in the Vert.x AMQP client).

Reply:
Yep. I realise you see this as a temporary fix I just figured it would have much the same effect, such that the unaffected bits dont see any change in behaviour now, or again later when the temp fix is removed and it goes back.

Reply:
Looks good